### PR TITLE
Add Sandstorm for Work documentation

### PR DESCRIPTION
**Pull request by paulproteus**
Requesting review & merge from @kentonv . Any feedback is welcome, either here on GitHub, or privately.

### CODE REVIEW

File: docs/administering/for-work.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+Once you've done this, you will see information about your feature key on the **For Work** tab. All
+Sandstorm features enabled by your feature key will be available to you in the most
+contextually-relevant location.
+
+### All about feature keys
+
+A feature key is a text file of this format:
+
+```
+--------------------- BEGIN SANDSTORM FEATURE KEY ----------------------
```
Comment (paulproteus) [inline docs/administering/for-work.md]: For those who are curious: This is a truncated version of my ~/.ssh/id_rsa.pub . :)

**Comment by paulproteus** [comment]
The purpose of this doc is:

- As a Sandstorm user, I may want to learn about what a feature key is, in more depth.

- As a Sandstorm.io employee, I want a quick reference on what the expected duration of feature keys is, so that I make requests to Kenton to generate feature keys with the right duration.

- As a non-profit, I want a place that indicates publicly that I might get a special deal, and how to request that special deal.

- As a Sandstorm core developer (whether or not I'm a Sandstorm.io employee), I want to know how to enable/disable feature keys.

**Comment by paulproteus** [comment]
The specific review I want from @kentonv is:

- Is the technical language in this doc accurate?

- Is the marketing-y phrasing reasonable enough to ship?

- Is it OK to explain that a few Sandstorm for Work features are not implemented yet? I imagine it is OK, but wanted to sanity-check that. I actually think it's valuable to have information somewhere about what is/isn't implemented yet.

- Is it OK if I list you as the point of contact for people who want testing keys for development?

- Is the way I described the special pricing for non-profits OK?

For @jadeqwang 

- Is it OK to have nonprofits use the same Sandstorm for Work form as everyone else? I figure it is, but thought I should check since that form goes to you.

I'd appreciate a speedy review/merge and can fix things tomorrow (Fri) as needed to make that a reality.

File: docs/administering/for-work.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+- Log into your Sandstorm server.
+
+- Make sure you have enabled [Sandstorm for Work on your server](#enabling-sandstorm-for-work).
+
+- Click **Admin Settings** within your Sandstorm server; this should take you to `/admin/settings`.
+
+You should now see a checkbox that allows you to enable LDAP login.
+
+### Features coming soon
```
Comment (kentonv) [inline docs/administering/for-work.md]: Technically SAML is still "coming soon".

File: docs/administering/for-work.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+We're still working on the following features:
+
+**Organization management.** Choose an LDAP group that, upon login, should automatically be given
```
Comment (kentonv) [inline docs/administering/for-work.md]: We already implement auto-upgrade, but I guess you're saying "auto-upgrade a specific group", but that's not LDAP-specific and "Organization management" seems like the wrong title.

You could say "**Advanced user management**: Choose who is allowed to use the server based on groups or other policies, rather that based only on domain."

But I'm not sure if this really deserves a bullet point. It seems more like a refinement of what we have already.

File: docs/administering/for-work.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+_user_ status on this Sandstorm server.
+
+**Group Management.** This will allow you to share a grain with everyone in an LDAP/Active Directory
```
Comment (kentonv) [inline docs/administering/for-work.md]: Group management is not LDAP-specific -- Google has groups too.

File: docs/administering/for-work.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+If you are developing Sandstorm, and you need to enable Sandstorm for Work features, you can get a
+testing feature key by contacting Kenton Varda. You must also can enable testing mode by setting
+`IS_TESTING=yes` in `sandstorm.conf` for testing keys to be considered valid. Note that enabling
+testing mode forfeits all security.
+
+### Special pricing
+
+We're grateful for all the work of non-profits and community groups. For that reason, non-profits,
+charities, and small unincorporated community groups can often receive feature keys for free. Please
+[contact us via the Sandstorm for Work page](https://sandstorm.io/business) and tell us one to two
```
Comment (kentonv) [inline docs/administering/for-work.md]: Do we want non-profits going through the form? I actually don't know.

File: docs/administering/for-work.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
+[Sandstorm for Work](https://sandstorm.io/business) is the name for a collection of features that
+are typically used by companies. Customers can buy a _feature key_ to enable these features on a
+Sandstorm server running at their own organization. Special pricing, sometimes including free
+feature keys, are available to [non-profits and community groups](#special-pricing). For pricing,
+please read the information on the [main website](https://sandstorm.io/business). As an
+implementation detail, Sandstorm is fully open-source software.
+
+For now, feature keys are typically 60-day trial keys. This gives you time to evaluate Sandstorm for
```
Comment (kentonv) [inline docs/administering/for-work.md]: The one I issued to square is 90-day, though I don't know that I'm informed enough to have an opinion.

**Comment by kentonv** [comment]
I wonder if we should include a note on the fact that it's still open source.

> Is it OK to explain that a few Sandstorm for Work features are not implemented yet?

Yes.

File: docs/administering/for-work.md (hunk 5 above)
Comment (ocdtrekkie) [inline docs/administering/for-work.md]: Does the form have a checkbox to indicate if someone is a non-profit? If you want them to contact you through the form, it should, so you know how to address them.

**Comment by jadeqwang** [comment]
> Is it OK to have nonprofits use the same Sandstorm for Work form as everyone else? I figure it is, but thought I should check since that form goes to you.

Yes, since there's a dropdown in the form to note that anyway.

File: docs/administering/for-work.md (hunk 6 above)
Comment (paulproteus) [inline docs/administering/for-work.md]: +0, 90 days it is

File: docs/administering/for-work.md (hunk 2 above)
Comment (paulproteus) [inline docs/administering/for-work.md]: Yeah, fair enough :) I will adjust accordingly.

File: docs/administering/for-work.md (hunk 3 above)
Comment (paulproteus) [inline docs/administering/for-work.md]: I actually was trying to describe the existing organization management feature here! I forgot it was already implemented. I'll adjust accordingly.

File: docs/administering/for-work.md (hunk 4 above)
Comment (paulproteus) [inline docs/administering/for-work.md]: I'll adjust accordingly.

FWIW I did realize that, I just thought it'd be clearer to use an LDAP example, but it's now clear to me that limiting the example to LDAP makes clarity worse.

File: docs/administering/for-work.md (hunk 5 above)
Comment (paulproteus) [inline docs/administering/for-work.md]: I checked, and it does. Good call, @ocdtrekkie

**Comment by paulproteus** [comment]
For @kentonv - you wrote:

> I wonder if we should include a note on the fact that it's still open source.

There was this sentence in the opening paragraph:

> As an implementation detail, Sandstorm is fully open source software.

I've adjusted it to the following:

> ### Open source
>
> The code for Sandstorm for Work is maintained in the Sandstorm open source project, under the Apache
> License 2.0. Feel free to read [the code in GitHub](https://github.com/sandstorm-io/sandstorm).

Honestly I'm OK either way. I don't think it's important to emphasize that Sandstorm for Work is open source. I think it's important to say it somewhere so that if people are trying to find this out, they can find it out.

**Comment by paulproteus** [comment]
I've made the changes that were requested, so I'm going to self-merge this now. If people have other thoughts, I'm happy to address them speedily.

**Comment by kentonv** [comment]
Regarding open source: my point was that I think some people will be confused about Sandstorm for Work being open source and also requiring a feature key, so I was thinking there should be a section explaining our model there. But I don't know, maybe it's superfluous.

**Comment by paulproteus** [comment]
Yeah - I am not sure exactly how much attention I want to draw to that.
What do you think of the current phrasing?

**Comment by kentonv** [comment]
I don't have a strong opinion.

**Pull request merged**